core/trino-parser/src/main/java/io/trino/sql/tree/LogicalExpression.java

```diff
@@ -41,66 +43,60 @@ public Operator flip()
     }
 
     private final Operator operator;
-    private final Expression left;
-    private final Expression right;
+    private final List<Expression> terms;
 
-    public LogicalBinaryExpression(Operator operator, Expression left, Expression right)
+    public LogicalExpression(Operator operator, List<Expression> terms)
     {
-        this(Optional.empty(), operator, left, right);
+        this(Optional.empty(), operator, terms);
     }
 
-    public LogicalBinaryExpression(NodeLocation location, Operator operator, Expression left, Expression right)
+    public LogicalExpression(NodeLocation location, Operator operator, List<Expression> terms)
     {
-        this(Optional.of(location), operator, left, right);
+        this(Optional.of(location), operator, terms);
     }
 
-    private LogicalBinaryExpression(Optional<NodeLocation> location, Operator operator, Expression left, Expression right)
+    private LogicalExpression(Optional<NodeLocation> location, Operator operator, List<Expression> terms)
     {
         super(location);
         requireNonNull(operator, "operator is null");
-        requireNonNull(left, "left is null");
-        requireNonNull(right, "right is null");
+        checkArgument(terms.size() >= 2, "Expected at least 2 terms");
 
         this.operator = operator;
-        this.left = left;
-        this.right = right;
+        this.terms = ImmutableList.copyOf(terms);
     }
 
     public Operator getOperator()
     {
         return operator;
     }
 
-    public Expression getLeft()
+    public List<Expression> getTerms()
     {
-        return left;
-    }
-
-    public Expression getRight()
-    {
-        return right;
+        return terms;
     }
 
     @Override
     public <R, C> R accept(AstVisitor<R, C> visitor, C context)
     {
-        return visitor.visitLogicalBinaryExpression(this, context);
+        return visitor.visitLogicalExpression(this, context);
     }
 
     @Override
     public List<Node> getChildren()
     {
-        return ImmutableList.of(left, right);
+        return terms.stream()
+                .map(Expression.class::cast)
+                .collect(toImmutableList());
```

Comment:
terms already are `Expression`, so the cast is no-op.

doing
```
return ImmutableList.copyOf(terms);
```

will be shorter and will change generic on the list as well